compiler/src/dmd/cparse.d

```diff
@@ -1915,6 +1915,7 @@ final class CParser(AST) : Parser!AST
                     {
                         auto str = asmName.peekString();
                         p.mangleOverride = str;
+                        p.adFlags |= AST.Declaration.nounderscore;
```

Comment:
The usual convention is to prefix the mangle string with an illegal asm character - gcc uses `*` - to disambiguate between regular symbol names and asm names.